activerecord/test/cases/relation/where_test.rb

```diff
@@ -336,6 +336,11 @@ def test_where_with_decimal_for_string_column
       assert_equal 0, count
     end
 
+    def test_where_with_rational_for_string_column
+      count = Post.where(title: Rational(0)).count
```

Comment:
Should this use the placeholder? Or have another assertion for the placeholder version?

Reply:
I'll add a whole group of them with the `"?"` syntax. There are none here.

Mostly I'm trying to add coverage with some `Rational`s because it's not a common use case but it's a valid one, and the behaviour can be pretty weird. (As we see with the division being done with the MySQL specific precision in the query)

Reply:
So it turns out there's an existing security bug I uncovered when adding coverage. #44408